### DISCUSSION TITLE
The most important feature missing in Semaphore

### DISCUSSION THREAD

**Discussion by fiftin** [semaphoreui/semaphore#709] [Ideas]
What's missing in Semaphore?

**Comment by MrChausson** [comment]
Task templates scheduling, this would be great to have.
It should display some warnings when modifying a task template / inventory which is already scheduled.

**Reply by fiftin** [reply]
Thank you. It will be implemented soon, just after vault/password feature.

**Reply by fiftin** [reply]
Task templates scheduling added:
https://github.com/ansible-semaphore/semaphore/releases/tag/v2.7.14

**Comment by scomcd** [comment]
We currently query the DB to generate a report showing scheduled tasks and their run times, but if possible, it would be great to have a view created within the GUI showing the Tasks that have been scheduled and their scheduled run times including date & time for next run etc

In the Task Templates view, it would be helpful to have an option that forces the alias string for each task to be unique. We use a sequenced alpha-numeric string as the alias for many of our tasks but the system currently allows duplicates.

**Comment by eisemakr** [comment]
hmm an improved user guide, would be great. There are many entries missing and in the wiki here many things are only very briefly described. I know this is an open source project :) and everyone like more to code than document but this would make it much more useful to others.

**Reply by fiftin** [reply]
Thank you, I agree, I working on docs. You can help the project by following it on [Twitter](https://twitter.com/AnsibleSem) and [StackShare](https://stackshare.io/ansible-semaphore).

**Reply by Freebase394** [reply]
Many Thanks! @fiftin and everyone ! 
This softwer its amazing! I'm testing it right now and i'm loving it.
I come here from Ansible AWX/AWX operator nightmare installations!

**Comment by steadfasterX** [comment]
from my point of view the following is required for serious usage of semaphore in companies but also for me as private person: https://github.com/ansible-semaphore/semaphore/issues/891

**Comment by jdhayes** [comment]
I agree with @steadfasterX, for enterprise users they will probably need better defined roles and perhaps LDAP authentication.

However, a more pressing feature for me is the API and how it could be extended.
I would really like to see an implementation of the "--limit" within the API so that the ansible run can be run exclusively on the calling host.

I am already running this in production with the following fork:
    https://github.com/jdhayes/semaphore

I have also made a new pull request:
    https://github.com/ansible-semaphore/semaphore/pull/811

**Reply by ansibleguy** [reply]
For limit see: [Feature - Limit as own Field](https://github.com/ansible-semaphore/semaphore/issues/1357)

**Comment by Freebase394** [comment]
Could someone add simple tooltip on app to show how to add and execute  simple commands to Inventory machines like "apt-get update" or "apt-get upgrade -y" please 
![image](https://github.com/ansible-semaphore/semaphore/assets/19652056/4af18d0d-2b33-43f8-bd40-2d9a3956c7db)

**Comment by s3lph** [comment]
Report when Ansible exits with changes!

This is a feature I'm currently missing from both AWX and Semaphore: Both only report whether a task succeeded or failed. Neither report whether a playbook run resulted in any changes. AWX has had [a corresponding issue](https://github.com/ansible/awx/issues/5564) for some years now, but without any activity AFAICT.

Implementation of such a feature doesn't sound too trivial. I currently see two ways how this can be achieved:
1. As suggested in the corresponding AWX issue: parsing the play recap section of the output.
2. Injecting a callback plugin into each run which would return the play recap in a form readable by Semaphore.

Unfortunately, neither of those sounds like a robust solution, as one involves parsing output intended for humans, and the other could interfere with other custom plugins.

**Comment by flyingfishflash** [comment]
A server->project level configuration property defining the length of time to keep the various types of generated data (activity/task execution history and output).

**Comment by derpedda** [comment]
Live-Tasklog would be amazing, reloading to see if the task succeeded is pretty painful (websockets are an option to achieve that)

**Reply by steadfasterX** [reply]
yea. it is working fine on the dashboard at least but not in the task template view

**Comment by allebone** [comment]
Direct Netbox Support for Inventory and Access to Rendered Content. 

Lots of shops out there working on IaC are using Netbox as the single source of truth for their network configurations. I am working to solidify that concept on my team, but with Semaphore as the Executor of that Vision. 

Having a direct and better connection to Netbox would allow regional/site specific semaphore installations to run against only their devices, and allow the non-coding crowd to focus on making Netbox look "true." Netbox could then run Webhooks targeted at Sempahore to execute a Task via the API, closing the loop. 

I can elaborate more, but it seems like a Win-Win for Netbox and Semaphore. 
[Semaphore + Netbox.pdf](https://github.com/ansible-semaphore/semaphore/files/13932470/Semaphore.%2B.Netbox.pdf)

**Reply by jdhayes** [reply]
Dynamic inventory is already an Ansible plugin away for Netbox source of truth.
I am using a custom script currently, but may change over to a more fully functional alternative. 

https://docs.ansible.com/ansible/latest/collections/netbox/netbox/nb_inventory_inventory.html

**Reply by allebone** [reply]
Oh for sure, I was just thinking for the bigger audience, adding a direct to Netbox Inventory configuration to the Sempahore GUI could eliminate a lot of the requirements.yml confusion. Plus to be honest, as a member of the Network Engineering crowd...it might spur a few to use Source of Truth in their life. :)

**Comment by fredix** [comment]
Hi all !
for me the biggest feature missing are the workflows : https://docs.ansible.com/ansible-tower/latest/html/userguide/workflows.html
Thanks for your excellent application.

**Reply by nickalleyne** [reply]
Very much agreed, this is the sort of thing that would allow a more broad audience to use the application.
